lib/rules/landmark-unique-matches.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Since this is a best-practice rule, we are filtering elements as dictated by ARIA 1.1 Practices regardless of treatment by browser/AT combinations.
+ *
+ * Info: https://www.w3.org/TR/wai-aria-practices-1.1/#aria_landmark
+ */
+const excludedParentsForHeaderFooterLandmarks = [
+	'article',
+	'aside',
+	'main',
+	'nav',
+	'section'
+].join(',');
+function isHeaderFooterLandmark(headerFooterElement) {
+	return !axe.commons.dom.findUpVirtual(
+		headerFooterElement,
+		excludedParentsForHeaderFooterLandmarks
+	);
+}
+
+function isLandmarkVirtual(virtualNode) {
+	const { actualNode } = virtualNode;
+	const landmarkRoles = axe.commons.aria.getRolesByType('landmark');
+	const role = axe.commons.aria.getRole(actualNode);
+	const tagName = actualNode.nodeName.toLowerCase();
```

Comment:
as a general practise with in `axe-core`, when using `node.nodeName`, we tend to always use `toUpperCase()`, please replace this with `const nodeName = actualNode.nodeName.toUpperCase()`.

We are looking to enhance our `eslint` rules to catch this at the moment. See https://github.com/dequelabs/axe-core/issues/1052 for more information.

You will have to update your usages as well `nodeName === 'HEADER'` and so on.

Reply:
@wahaj reason for this is XHTML documents

Reply:
Done 😄